src/Opaleye/Internal/TypeFamiliesTH.hs

```diff
@@ -0,0 +1,184 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Opaleye.Internal.TypeFamiliesTH (makeTableAdaptorInstanceMap) where
+
+import Data.List (foldl', nub)
+import Data.Maybe (listToMaybe, mapMaybe)
+import Data.Profunctor.Product (ProductProfunctor)
+import Data.Profunctor.Product.Default (Default (..))
+import Data.Profunctor.Product.Internal.TH (adaptorDefinition, adaptorDefinitionFields)
```

Comment:
This can't depend on stuff in an internal module from a different package.  If it really needs these I'll have to expose them from product-profunctors.

Reply:
@tomjaguarpaw I made a PR to `product-profunctors`

Reply:
Cool, thanks!